masto-module/src/main/kotlin/au/rakka/java/masto-api/MastoNotifier.kt

```diff
@@ -40,29 +39,32 @@ public class MastoNotifier @Inject constructor(private val conf:MastoConfig) : G
 	override suspend fun notifyNewGame(notificationId: String, game: Game)
 	{
 		logger.debug("Notifying new game")
-		hostedGameTags[game.id]=notificationId
-		val response = client.sendAsync(builder.uri(URI.create(statusurl)).POST(game_to_string(game,hostedGameTags[game.id])).build(),HttpResponse.BodyHandlers.ofString()).await()
+		val response = client.sendAsync(builder.uri(URI.create(statusurl)).POST(game_to_string(game,notificationId)).build(),HttpResponse.BodyHandlers.ofString()).await()
 		logger.debug(response.statusCode().toString())
 		val node=mapper.readTree(response.body())
 		logger.debug(response.body())
-		hostedGameMessages[game.id]=node.get("id").asText()
-		logger.debug("Saved ID {}: {}",game.id,hostedGameMessages[game.id])
+		hostedGames[game.id]=GameMessage(node.get("id").asText(),notificationId,game.name,game.currentPlayers)
+		logger.debug("Saved ID {}: {}",game.id,hostedGames[game.id]!!.message_id)
 	}
 	
 	override suspend fun updateExistingGame(game: Game)
 	{
-		logger.debug("Updating {}",hostedGameMessages[game.id])
-		val response=client.sendAsync(builder.uri(URI.create(statusurl+"/"+hostedGameMessages[game.id])).PUT(game_to_string(game,hostedGameTags[game.id])).build(),HttpResponse.BodyHandlers.ofString()).await()
+		logger.debug("Update function called")
+		val gm=hostedGames[game.id]!!
+		logger.debug("Might update {}",gm.message_id)
+		if (!gm.update(game)) { return }
```

Comment:
You don't actually need curly braces here, and it might read better without them. `if (!gm.update(game)) return`